### PR TITLE
Use https for sourceforge during CI

### PR DESCRIPTION
**Pull request by dns2utf8**
I saw that we use http during CI opening up the CI process to on the wire tampering.

based on #86573 

r? @pietroalbini

### CODE REVIEW

**Comment by dns2utf8** [comment]
I followed the new upstream of expat to github in the second commit. If you prefer to stay on sourceforge I could remove the second commit. However, I think we should follow the project and maybe have another mechanism to track upstream changes.

Cheers,
Stefan

**Comment by pietroalbini** [comment]
Thanks!

@bors r+

File: src/ci/docker/host-x86_64/dist-riscv64-linux/riscv64-unknown-linux-gnu.config [hunk 1]
```diff
@@ -645,7 +645,8 @@ CT_EXPAT_PATCH_ORDER="global"
 CT_EXPAT_V_2_2=y
 # CT_EXPAT_NO_VERSIONS is not set
 CT_EXPAT_VERSION="2.4.1"
-CT_EXPAT_MIRRORS="http://downloads.sourceforge.net/project/expat/expat/${CT_EXPAT_VERSION}"
+CT_EXPAT_VERSION_TAG="2_4_1"
+CT_EXPAT_MIRRORS="https://github.com/libexpat/libexpat/releases/download/R_${CT_EXPAT_VERSION_TAG}/expat-${CT_EXPAT_VERSION}.tar.bz2"
```
Comment (pietroalbini) [inline src/ci/docker/host-x86_64/dist-riscv64-linux/riscv64-unknown-linux-gnu.config:649]:
```suggestion
CT_EXPAT_MIRRORS="https://github.com/libexpat/libexpat/releases/download/R_${CT_EXPAT_VERSION_TAG}"
```

Comment (dns2utf8) [reply src/ci/docker/host-x86_64/dist-riscv64-linux/riscv64-unknown-linux-gnu.config]: Thanks!

**Pull request merged**